examples/bert/run_glue_finetuning.py

```diff
@@ -200,28 +198,60 @@ def preprocess_data(inputs, labels):
 
     # Read and preprocess GLUE task data.
     train_ds, test_ds, validation_ds = load_data(FLAGS.task_name)
-    train_ds = train_ds.batch(FLAGS.batch_size).map(
+
+    # batch_size is taken as 32.
+    train_ds = train_ds.batch(batch_size).map(
         preprocess_data, num_parallel_calls=tf.data.AUTOTUNE
     )
-    validation_ds = validation_ds.batch(FLAGS.batch_size).map(
+    validation_ds = validation_ds.batch(batch_size).map(
         preprocess_data, num_parallel_calls=tf.data.AUTOTUNE
     )
-    test_ds = test_ds.batch(FLAGS.batch_size).map(
+    test_ds = test_ds.batch(batch_size).map(
         preprocess_data, num_parallel_calls=tf.data.AUTOTUNE
     )
 
-    finetuning_model = BertClassificationFinetuner(
-        bert_model=model,
-        hidden_size=bert_config["hidden_size"],
-        num_classes=3 if FLAGS.task_name in ("mnli", "ax") else 2,
+    class BertHyperModel(keras_tuner.HyperModel):
```

Comment:
We had a discussion [in this comment](https://github.com/keras-team/keras-nlp/pull/143#discussion_r859054948) which stated that `__init__` should be removed so as to not cause problems, but it was mainly because of passing `model` into `HyperModel`.
I'll change it in the next commit.

Reply:
We could actual recreate the entire model in `build` right? E.g. don't pass the model in, but rather load the encoder model from disk, and recreate the BertClassificationFinetuner in each build call.

That might be the correct thing to do here, as we don't want to share weights between fine-tuning runs in the original encoder.

Reply:
Yeah, that makes sense. I suppose we can move the entire upper portion where we initialize the model, in the `main()` function to the `build()` method.

Reply:
Sorry for being late, I got a bit wound up with college stuff, but coming back to this, do we need an `__init__` in the class? Or are we proceeding with how [the discussion went here](https://github.com/keras-team/keras-nlp/pull/143#discussion_r859182295).

Reply:
I believe (might be wrong here) that because the base encoder model is only loaded once, and the model that is recreated here uses the same encoder model, the same weights will be used for the underlying encoder across all fine tuning runs. That is incorrect if it is happening! We need to recreate the entire model including loading the encoder I think.

And from a readability perspective this class we be better off split out instead on inlined.

So yes, I think an __init__ and moving this out makes sense.

Reply:
Yeah, that makes sense. So, I'll pass the `bert_config` as an argument to the `__init__` because computing that each time wouldn't make sense I think.